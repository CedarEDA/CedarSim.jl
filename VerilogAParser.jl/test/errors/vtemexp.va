module vtemexp(p, n);

    analog function real vtmexp;
        input VTM0, vjct, njtemp, VTS;
        real VTM0, vjct, njtemp, VTS;

        begin
            Nvtm = VTM0 * njtemp;
            if ((VTS - vjct) < VTS * 1e-3) begin
                T9 = 1.0e3;
                T0 = - vjct / Nvtm * T9;
                vtmexp = lexp(T0);
            else
                T9 = 1.0 / (VTS - vjct);
                T0 = - vjct / Nvtm * VTS * T9;
                vtmexp = lexp(T0);
            end
        end
    endfunction

    inout p, n;
    electrical p, n;

    analog begin
        I(p,n) <+ 0;
    end
endmodule
