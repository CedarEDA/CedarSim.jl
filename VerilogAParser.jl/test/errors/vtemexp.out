[31m[1m╭ ERROR [VAS49]:[22m[39m[31m Unterminated Analog Block.[39m
[31m│[39m[90m   ╭───── vtemexp.va:9[39m
[31m│[90m9[0m  [90m│[0m             if ((VTS - vjct) < VTS * 1e-3) begin
[31m│[0m   [90m┊[0m                                            [33m━━┳━━
[31m│[0m   [90m┊[0m                                      [33mBlock starts here[0m
[31m│[90m10[0m [90m│[0m                 T9 = 1.0e3;
[31m│[90m11[0m [90m│[0m                 T0 = - vjct / Nvtm * T9;
[31m│[90m12[0m [90m│[0m                 vtmexp = lexp(T0);[91m◌ 
[31m│[0m   [90m┊[0m                                   [32mend[0m
[31m│[90m13[0m [90m│[0m             else
[31m│[39m[90m   │[39m
[31m│[39m[90m   ╰───── [39m